ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,296 @@
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Third-party Imports
+from boto.ec2.elb.healthcheck import HealthCheck
+import boto.exception
+
+# Cloudify imports
+from ec2 import constants
+from ec2 import connection
+from ec2 import utils
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+
+@operation
+def create_health_check(**_):
+    _create_health_check()
+
+
+@operation
+def create_load_balancer(**_):
+
+    lb = _create_load_balancer()
+
+    health_check = ctx.node.properties.get('health_check')
+
+    if isinstance(health_check, list):
+        if health_check:
+            _add_health_check_to_load_balancer(lb, health_check[0])
+
+
+def _add_instance_to_load_balancer_list_in_properties(instance_id):
+
+    if 'instance_list' not in ctx.target.instance.runtime_properties.keys():
+        ctx.target.instance.runtime_properties['instance_list'] = []
+
+    ctx.target.instance.runtime_properties['instance_list'].append(instance_id)
+
+
+def _remove_instance_from_load_balancer_list_in_properties(instance_id):
+    if instance_id in ctx.target.instance.runtime_properties['instance_list']:
+        ctx.target.instance.runtime_properties['instance_list'].remove(
+            instance_id)
+
+
+@operation
+def remove_instance_from_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to remove_instance_from_load_balancer.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    instance_list = [instance_id]
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.get_all_load_balancers(
+            load_balancer_names=[elb_name])[0]
+        lb.deregister_instances(instance_list)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Instance not removed from Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_load_balancer_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to add_instance_to_load_balancer.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.get_all_load_balancers(
```

Comment:
move to 2 methods:
1) get a list of existing load balancers that match given list of ids
if not found does another call without params and logs the complete list of elbs and then returns none
otherwise returns the list
2) calls (1) and returns first matching unless None.